### ISSUE TITLE
Address name conflicts

### ISSUE THREAD

**Issue by stevendborrelli** [asteris-llc/converge#361]
If two resources have the same name an error is not raised. In testing the 2nd task defined is used. 

```
task "a" {
   check = "[[ -d /tmp/a ]]"
   apply = "mkdir /tmp/a"
}

task "a" {
   check = "[[ -d /tmp/b ]]"
   apply = "mkdir /tmp/b"
}
```

**Comment by ryane** [comment]
This hcl also can also result in some user confusion:

```hcl
task "c" {
  check = "[[ -d /tmp/c ]]"
  apply = "mkdir /tmp/c"
}

task "d" {
  check = "[[ -d /tmp/c ]]"
  apply = "mkdir /tmp/c"
}
```

It is the same task with different names but you'll never see the duplicate in your graph.

`MergeDuplicates` explicitly handles both of these scenarios and you can identify when it happens if you are running with `log-level=debug`. Should we just change the logging to Warn with a clearer message or should we actually consider one or both of these as errors and stop execution?

**Comment by arichardet** [comment]
Fixed by #458
